### PR TITLE
Add C++ doc links on type doc page

### PR DESCRIPTION
**Pull request by Wumpf**
### What

* Fixes https://github.com/rerun-io/rerun/issues/4332

Also, now using the python versioning scheme instead of the rust versioning scheme for docs on gh-pages branch.

The Python one seems to keep versions nicely which we leave to docs.rs on Rust
* Python: https://github.com/rerun-io/rerun/tree/gh-pages/docs/python
* Rust: https://github.com/rerun-io/rerun/tree/gh-pages/docs/rust

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/4339) (if applicable)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG

- [PR Build Summary](https://build.rerun.io/pr/4339)
- [Docs preview](https://rerun.io/preview/57a74625dc5b5015748c224da6b382923361ec71/docs) 
- [Examples preview](https://rerun.io/preview/57a74625dc5b5015748c224da6b382923361ec71/examples) 
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

### CODE REVIEW

File: crates/re_types_builder/src/codegen/docs/mod.rs [hunk 1]
```diff
@@ -173,6 +172,16 @@ fn object_page(reporter: &Reporter, object: &Object, object_map: &ObjectMap) ->
             object.kind.plural_snake_case(),
             object.name
         );
+        putln!(
+            page,
+            // `_1` is doxygen's replacement for ':'
+            // https://github.com/doxygen/doxygen/blob/Release_1_9_8/src/util.cpp#L3532
+            " * 🌊 [C++ API docs for `{}`](https://ref.rerun.io/docs/cpp/stable/structrerun_1_1{}_1_1{}.html{})",
```
Comment (emilk) [inline crates/re_types_builder/src/codegen/docs/mod.rs:179]: Let's sort the languages lexicographically instead of… whatever order this is :)

Comment (emilk) [reply crates/re_types_builder/src/codegen/docs/mod.rs:179]: The order is also weird here:

image

**Review by emilk — APPROVED** [review]
Great!

There are a few other places that should have the C++ API docs linked:
* `/README.md`
* `rerun_cpp/README.md`
* `BUILD.md` section `Building the docs`
* ???

File: crates/re_types_builder/src/codegen/docs/mod.rs (hunk 1 above)
Comment (Wumpf) [reply crates/re_types_builder/src/codegen/docs/mod.rs:179]: Python, C++, Rust is the order we use pretty much everywhere

Comment (emilk) [reply crates/re_types_builder/src/codegen/docs/mod.rs:179]: Far from everywhere:

image

Screenshot 2023-11-27 at 12 27 41

image

Comment (emilk) [reply crates/re_types_builder/src/codegen/docs/mod.rs:179]: As the number of supported languages grows, anything but lexicographical order will just be weird

**Comment by emilk** [comment]
Is this closing https://github.com/rerun-io/rerun/issues/4332 btw?

**Comment by Wumpf** [comment]
yes!

**Comment by Wumpf** [comment]
added doc links in the other two places but `rerun_cpp/README.md` is a weird one because it is also the landing page of the docs itself 🤔

**Pull request merged**